### PR TITLE
Added new cmdlets which check the build environment and stop tasks in the event of an error

### PR DESCRIPTION
**Pull request by russellseymour**
## 📲 What

Fixed issue with the tests coverage being completed successfully.

Added `Confirm-Environment` that checks all the required environment variables have been set, according to a settings file in the repository.
Added `Stop-Task` which throws an exception which makes `taskctl` break out of the pipeline

Updated documentaion

## 🤔 Why

The task for running the Unit test coverage was failing due to the output of Pester including characters that look like Golang templating

A lot of the configuration in the independent runner can be set using enviornment variables, however it is not obvious what these need to be. The new `Confirm-Environment` cmdlet performs this task and fails when env vars are missing and states which ones need to be set.

When tasks were failing previously, `taskctl` did not break out of the pipeline. This mean that sometimes the pipeline appeared to have succeeded but it had failed.

Most of the cmdlets in the module have help that can be accessed in PowerShell using `Get-Help <Name>` but as we generating documentation this should be included as well.

## 🛠 How

Changed the verbosity of the Invoke-Pester from Detailed to Normal so that `taskctl` does not see the offending string.

When the `Confirm-Parameters` cmdlet finds that an specific environment variable does not exist, it calls the `Stop-Task` cmdlet which throws an exception which causes the pipeline to fail.

The tasks now call a new script`Build-Help.ps1` which iterates around the exported functions and gets the help for each one. This is then saved to a file and the main documentation is then tun.

## 👀 Evidence

The following image shows the `Confirm-Environment` command running and stating the envs that are missing and then calling the `Stop-Task` command to stop the pipeline.

image

The following image shows the command help added to the documentation

image

## 🕵️ How to test

Notes for QA

### CODE REVIEW

File: build/azuredevops-runner.yml [hunk 1]
```diff
@@ -118,6 +127,15 @@ stages:
...
 
+        # Install Stacks Envfile to generate the envfile to be used by docker
+        - task: Bash@3
+          displayName: Install Envfile
+          inputs:
+            targetType: inline
+            script: |
+              wget https://github.com/amido/stacks-envfile/releases/download/v${{ variables.StacksEnvfileVersion }}/stacks-envfile-linux-amd64-${{ variables.StacksEnvfileVersion }} -O /usr/local/bin/envfile
+              chmod +x /usr/local/bin/envfile
+
```
Comment (williamayerst) [inline build/azuredevops-runner.yml:138]: Are we installing Envfile twice?

File: build/azuredevops-runner.yml [hunk 2]
```diff
@@ -129,4 +147,3 @@ stages:
             env: 
               PUBLISH_RELEASE: $true
               DASHBOARD_INFLUX_TOKEN: $(DASHBOARD_INFLUX_TOKEN) # requires explicit mapping to be used as an env var
```
Comment (williamayerst) [inline build/azuredevops-runner.yml:149]: Missing newline

File: src/modules/AmidoBuild/exported/Invoke-Terraform.Tests.ps1 [hunk 3]
```diff
@@ -130,6 +136,19 @@ Describe "Invoke-Terraform" {
             Invoke-Terraform -plan -Path $testFolder -arguments "-input=false,-out=tfplan"
             $Session.commands.list[0] | Should -BeLike "*terraform* plan -input=false -out=tfplan"
         }
+
+        It "will will run the the plan command using an environment variable for the arguments" {
```
Comment (williamayerst) [inline src/modules/AmidoBuild/exported/Invoke-Terraform.Tests.ps1:140]: will will? :)

File: build/azuredevops-runner.yml (hunk 1 above)
Comment (russellseymour) [reply build/azuredevops-runner.yml:138]: Yeah we are, my bad - thought we needed to but of course it is all in the same agent, will fix that one

File: src/modules/AmidoBuild/exported/Invoke-Terraform.Tests.ps1 (hunk 3 above)
Comment (russellseymour) [reply src/modules/AmidoBuild/exported/Invoke-Terraform.Tests.ps1:140]: OK, will sort

File: build/azuredevops-runner.yml (hunk 1 above)
Comment (russellseymour) [reply build/azuredevops-runner.yml:138]: As this is a differennt stage in the pipeline it is run in a different session so the tools that were installed during the build stage are not available, this means that `taskctl` and `envfile` need to be installed int he Release stage as well

**Review by williamayerst — APPROVED** [review]

**Pull request merged**